### PR TITLE
PreviewOptions: Fix critical error when children not passed

### LINKED ISSUE #54283: PreviewOptions: critical error when children not passed
### Description

It was also reported in:

- Core Ticket: https://core.trac.wordpress.org/ticket/59317
- pattern-directory repo: https://github.com/WordPress/pattern-directory/issues/598

The PreviewOptions component renders child components. However, because it assumes that the child is a function, a critical error will occur if the child is not passed in the proper form.

In the pattern directory, [no children are passed](https://github.com/WordPress/pattern-directory/blob/d3c6c3618c50827d46e9101236984c06ebcd5419/public_html/wp-content/plugins/pattern-creator/src/components/header/index.js#L118), as only a device preview is sufficient.

### Step-by-step reproduction instructions

To reproduce this issue, rewrite the Edit component of the code block as follows:

```javascript
/**
 * WordPress dependencies
 */
import {
	useBlockProps,
	__experimentalPreviewOptions as PreviewOptions,
} from '@wordpress/block-editor';
import { useDispatch, useSelect } from '@wordpress/data';

export default function CodeEdit() {
	const deviceType = useSelect( ( select ) => {
			return select( 'core/edit-post' ).__experimentalGetPreviewDeviceType();
	}, [] );
	const { __experimentalSetPreviewDeviceType: setPreviewDeviceType } =
		useDispatch( 'core/edit-post' );
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<PreviewOptions deviceType={ deviceType } setDeviceType={ setPreviewDeviceType } />
		</div>
	);
}
```

- Insert a code block.
- The block should break.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #54283

## What?

This PR fixes a critical error that occurs when the PreviewOptions component has no children.

## Why?

This is because this component expects to be passed a child that is a function. This component should not require children. Even the `PreviewOptions` component in the pattern-directory [only wants the device type dropdown](https://github.com/WordPress/pattern-directory/blob/d3c6c3618c50827d46e9101236984c06ebcd5419/public_html/wp-content/plugins/pattern-creator/src/components/header/index.js#L118).

## How?

Add `typof` check and Update document.

## Testing Instructions

Rewrite the Code block with a PreviewOptions component that has no children as below:

```javascript
/**
 * WordPress dependencies
 */
import {
	useBlockProps,
	__experimentalPreviewOptions as PreviewOptions,
} from '@wordpress/block-editor';
import { useDispatch, useSelect } from '@wordpress/data';

export default function CodeEdit() {
	const deviceType = useSelect( ( select ) => {
			return select( 'core/edit-post' ).__experimentalGetPreviewDeviceType();
	}, [] );
	const { __experimentalSetPreviewDeviceType: setPreviewDeviceType } =
		useDispatch( 'core/edit-post' );
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<PreviewOptions deviceType={ deviceType } setDeviceType={ setPreviewDeviceType } />
		</div>
	);
}
```

- Insert a Code block.
- trunk: The block should break.
- This PR: The dropdown should now work properly.

## Screenshots or screencast 

preview-options

### CODE REVIEW

File: packages/block-editor/src/components/preview-options/README.md [hunk 1]
```diff
@@ -82,6 +82,13 @@ Used to set the device type that will be used to display the preview inside the
 -   Type: `func`
 -   Required: yes
 
+#### children
+
+A function that returns nodes to be rendered within the dropdown.
+
+-   Type: `Function`
+-   Required: No
+
```
Comment (t-hamano) [inline packages/block-editor/src/components/preview-options/README.md:91]: This definition was inspired by [the AutoComplete component](https://github.com/WordPress/gutenberg/blob/b606951aac8e513bd48022b04c0c002978fd87ad/packages/components/src/autocomplete/README.md?plain=1#L44-L49).

File: packages/block-editor/src/components/preview-options/index.js [hunk 2]
```diff
@@ -81,7 +81,8 @@ export default function PreviewOptions( {
 							{ __( 'Mobile' ) }
 						</MenuItem>
 					</MenuGroup>
-					{ children( renderProps ) }
+					{ typeof children === 'function' &&
+						children( renderProps ) }
```
Comment (ntsekouras) [inline packages/block-editor/src/components/preview-options/index.js]: Nit: I guess we could just `{ children?.( renderProps ) }`

**Review by ntsekouras — APPROVED** [review]
Looks good, thank you!

File: packages/block-editor/src/components/preview-options/index.js (hunk 2 above)
Comment (t-hamano) [reply packages/block-editor/src/components/preview-options/index.js]: Makes sense! I have confirmed that the optional chaining also works as expected.

**Pull request merged**